### ISSUE TITLE
Provided dependencies from imported BOM are not correctly calculated on hover

### ISSUE THREAD

**Issue by CarlosMOGoncalves** [eclipse/lemminx-maven#347]
# Description #
----------

Hello there, 

In **Eclipse IDE for Enterprise Java and Web Developers 2022-12** the provided dependencies that are managed by a BOM import declared in its POM or a parent POM are not correctly calculated.

The editor, on hover, will state that the managed on which is not the one declared on the import dependency, but some other (I haven't figured which one it is picking).

If, however, we `Ctrl+click` on it, to navigate to the POM, it will navigate to the correct version.

This seems to happen only on **provided** dependencies and **not on compile**, as the latter are correctly calculated when the imported BOM is present.

## Steps to reproduce ## 

I have prepared a sample application that has a very simple Payara Micro application
It can be found [here](https://github.com/CarlosMOGoncalves/eclipse-issue-tester)

This app does nothing special. It has a Jakarta REST endpoint but that is just for show.

1. Fresh install Eclipse for **Eclipse IDE for Enterprise Java and Web Developers 2022-12**
2. Checkout the project from my repository
3. Proceed to build the project using `mvn clean install`
4. Navigate to the POM and hover over one of the provided dependencies, for example the Jakarta one:

![image](https://user-images.githubusercontent.com/2636004/212747372-33fd8d72-fcd2-4949-8917-dbf9bb7fbfd1.png)

5. This is not the correctly managed one, as that is stated on the BOM imported above and its value is **8.0.0**
6. If we `Ctrl+click` the jakartaee-api dependency it will navigate to the correct **8.0.0** pom
7. This happens in all those 3 **provided** dependencies: jakartaee, microprofile and eclipselink
8. However, it does not happen when we check the **compile** dependency, in this case AWS SDK, which is the correct one, as imported above:

![image](https://user-images.githubusercontent.com/2636004/212747910-ba933e6a-2889-4ac6-8703-9ad7085c54d6.png)

9. Also, I think that the poms where these dependencies are **managed** might not be correct. I think all those dependencies are managed in the respective BOMs above, but when `Ctrl-clicking` them, it should navigate to the **artifact POM**, as it is doing right now.

## Environment ##
- **Distribution**:  Eclipse IDE for Enterprise Java and Web Developers 2022-12 2022-12 for Windows x64
- **M2E version**:  2.1.2.20221130-2239 (eclipse 2022-12 default)

**Comment by mickaelistria** [comment]
Do you think you could create a minimal reproducer for this issue (with 1 parent BOM and 1 consumer and just 1 dependency) in order to facilitate debug and even to include it as a test case directly.

> JDK Version: Corretto JDK 11

In practice, your IDE is using the embedded OpenJDK 17 packaged for Eclipse RCP applications by Eclipse JustJ project.

**Comment by CarlosMOGoncalves** [comment]
Sure, let's give it a try.

I have created this project in my repo, [eclipse-bom-tester](https://github.com/CarlosMOGoncalves/eclipse-bom-tester)

It has a **child pom** that has a **parent pom** which resides inside the parent folder.

Although you asked me for just one dependency I have included 2 because I found another issue that is likely related to this one.

In short:

- dependencies whose origin reside in an **import** pom will have their version incorrect. Below, the correct version as imported should be **8.0.0**, **not 9.0.0-RC2**
- 
![image](https://user-images.githubusercontent.com/2636004/212890787-a61552b8-7b74-4255-b7b1-4a39efe9b805.png)

- dependencies whose origin reside in the parent pom AND their version **is defined in a property** will rather show the property name and not the actual number

![image](https://user-images.githubusercontent.com/2636004/212891335-57242e16-d909-44ed-9890-f6338f94f117.png)

- In both cases,` Ctrl-click` will navigate to the correct version

> In practice, your IDE is using the embedded OpenJDK 17 packaged for Eclipse RCP applications by Eclipse JustJ project.

You are correct. This is unneeded info I just copied from another issue of mine, where the version is relevant and I am using an external JDK for runtime. I will remove it from he original issue.
